src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -508,12 +508,11 @@ public function createSchemaTable()
 
             $table = new Table($this->getSchemaTableName(), $options, $this);
 
-            if ($this->getConnection()->getAttribute(\PDO::ATTR_DRIVER_NAME) === 'mysql'
-                && version_compare($this->getConnection()->getAttribute(\PDO::ATTR_SERVER_VERSION), '5.6.0', '>=')) {
```

Comment:
Isn't this `version_compare` condition trying to solve the exact issue you are describing? It looks like the baseline would be get worse for everyone with this change, not just users on MySQL < 5.6.